mne_connectivity/decoding/decomposition.py

```diff
@@ -0,0 +1,472 @@
+                raise ValueError(
+                    "at least one entry in `rank` is greater than the number of "
+                    "seed/target channels in `indices`"
+                )
+
+        return rank
+
+    def fit(self, X, y=None):
+        """Compute connectivity decomposition filters for epoched data.
+
+        Parameters
+        ----------
+        X : array, shape=(n_epochs, n_signals, n_times)
+            The input data which the connectivity decomposition filters should be fit
+            to.
+        y : None
+            Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        self : instance of CoherencyDecomposition
+            The modified class instance.
+        """
+        # validate input data
+        self._check_X(X, ndim=[3])
+        self._get_rank_and_ncomps_from_X(X)
+
+        # compute CSD
+        csd = self._compute_csd(X)
+
+        # instantiate connectivity estimator and add CSD information
+        self._conn_estimator = self._conn_estimator(
+            n_signals=X.shape[1],
+            n_cons=1,
+            n_freqs=1,
+            n_times=0,
+            n_jobs=self.n_jobs,
+            store_filters=True,
+        )
+        self._conn_estimator.accumulate(con_idx=np.arange(csd.shape[0]), csd_xy=csd)
+
+        # fit filters to data and compute corresponding patterns
+        self._conn_estimator.compute_con(
+            indices=self._indices, ranks=self._rank, n_epochs=1
+        )
+
+        # extract filters and patterns
+        self._extract_filters_and_patterns()
+
+        return self
+
+    def _check_X(self, X, ndim):
+        """Check that the input data is valid."""
+        # check data is a 2/3D array
+        _validate_type(X, np.ndarray, "`X`", "NumPy array")
+        _check_option("`X.ndim`", X.ndim, ndim)
+        n_chans = X.shape[-2]
+        if n_chans != self.info["nchan"]:
+            raise ValueError(
+                "`X` does not match Info\nExpected %i channels, got %i"
+                % (n_chans, self.info["nchan"])
+            )
+
+    def _get_rank_and_ncomps_from_X(self, X):
+        """Get/validate rank and n_components parameters using the data."""
+        # compute rank from data if necessary / check it is valid for the indices
+        rank = _check_rank_input(self._rank, X, self._indices)
+
+        # set n_components if necessary / check it is valid for the rank
+        if self.n_components is None:
+            self.n_components = np.min(rank)
+        elif self.n_components > np.min(rank):
+            raise ValueError(
+                "`n_components` is greater than the minimum rank of the data"
+            )
+
+        # set rank if necessary
+        if self._rank is None:
+            self._rank = rank
+
+    def _compute_csd(self, X):
+        """Compute the cross-spectral density of the input data."""
+        # XXX: fix csd returning [fmin +1 bin to fmax -1 bin] frequencies
```

Comment:
is this `XXX` comment meant to be addressed in this PR?

Reply:
This has been solved, just forgot to remove. Thanks for pointing it out!